firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/Bytes.txt

```diff
@@ -0,0 +1,59 @@
+Scenario:The Util get_byte function correctly gets bytes from the given position
+   GIVEN("The util get_byte function")
+       WHEN("the get_byte function is called on 0x86 ")
+          THEN(""The value at the least significant byte position is "0x0a'")
+   GIVEN("The util get_byte function " )
+       WHEN("the get_byte function is called on 0x860a")  
+       THEN(""The value at the least significant byte position is "0x0a")
+       THEN("The value at the most signficant postiion is 0x86")
+
+
+   GIVEN("The util get_byte function'")
+       WHEN("the get_byte function is called on 0x86010f0a")
+       THEN("value is as expected ")
+
+
+Scenario: The Util set_byte function correctly sets bytes to the given position
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x86 for 0th index")
+     THEN("Value is as expected 134 ")
+
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x86 for 0th index")
+     THEN("Value is as expected 0 ")
+
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x83 for 0th index")
+     THEN("Value is as expected in uint16_t data-type ")
+  
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x83 for 1st index")
+     THEN("Value is as expected in uint16_t data-type ")
+     
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x83 for 2nd index")
+     THEN("Value is as expected in uint32_t data-type ")
+
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x83 for 3rd index")
+     THEN("Value is as expected in uint32_t data-type ")
+
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x860a for 0th index")
+     THEN("Value is as expected in uint16_t data-type ")
+
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x860a for 1st index")
+     THEN("Value is as expected in uint16_t data-type ")
+
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x860a for 2nd index")
+     THEN("Value is as expected in uint16_t data-type ")
+
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x860a for 2nd index")
+     THEN("Value is as expected in uint32_t data-type ")
+
+  GIVEN("The Util set_byte function ")
+  WHEN("the get_byte function is called on 0x860a for 3rd index")
```

Comment:
The `WHEN`s should specify when we're calling `set_byte` with byte_index specified as a template parameter vs. when we're calling it with byte_index specified as a regular input parameter. See also my comment on the `get_byte` tests about needing to specify all template parameters. So these `WHENs` should look like e.g. `WHEN('The function is called with byte index 1 and output type uint16_t as the template parameters and 0x83 as the input value')`